### PR TITLE
Newsletters: add 268 (2023-09-13)

### PR DESCRIPTION
**Pull request by harding**
- [x] Lede, releases/RCs, topic entries @harding 
- [x] Bitcoin Core PR Review Club @LarryRuane 
- [x] Bitcoin Core 26567 @murchandamus

### CODE REVIEW

File: _topics/en/client-side-validation.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+that Bob controls.
+
+**[RGB][]** is a client-side validation protocol that uses
+[pay-to-contract][topic p2c] to allow transactions to commit to
+additional data, such as transfers.  The protocol has been designed to
+be highly flexible.
+
+**[Taproot Assets][]**, formerly called **Taro**, is a protocol heavily
+inspiried by RGB that uses [taproot][topic taproot]'s commitment
+structure to make allow transactions to commit to additional data.
```
Comment (bitschmidty) [inline _topics/en/client-side-validation.md:76]:
```suggestion
structure to allow transactions to commit to additional data.
```

File: _topics/en/client-side-validation.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+Taproot's construction itself derives from pay-to-contract.  As the name
+suggests, initial protocol developemnt is specifically focused on the
```
Comment (bitschmidty) [inline _topics/en/client-side-validation.md:78]:
```suggestion
suggests, initial protocol development is specifically focused on the
```

File: _posts/en/newsletters/2023-09-13-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    Sander's summary had not received any comments as of this writing.
+
+## Bitcoin Core PR Review Club
+
+*In this monthly section, we summarize a recent [Bitcoin Core PR Review
+Club][] meeting, highlighting some of the important questions and
+answers.  Click on a question below to see a summary of the answer from
+the meeting.*
+
+[transport abstraction][review club 28165] is a PR by Pieter Wuille (sipa)
```
Comment (bitschmidty) [inline _posts/en/newsletters/2023-09-13-newsletter.md:60]:
```suggestion
[Transport abstraction][review club 28165] is a PR by Pieter Wuille (sipa)
```

File: _posts/en/newsletters/2023-09-13-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+that introduces a _transport_ absraction (interface class). Concrete
```
Comment (bitschmidty) [inline _posts/en/newsletters/2023-09-13-newsletter.md:61]:
```suggestion
that introduces a _transport_ abstraction (interface class). Concrete
```

File: _posts/en/newsletters/2023-09-13-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+derivations of this class convert a (per-peer) connection's (already
+serialized) send and receive messages to and from wire format. This
+can be thought of as implementing a deeper level of serialization and
+deserialization. These classes do not do the actual sending and receiving.
+
+The PR derives two concrete classes from the `Transport` class,
+`V1transport` (what we have today) and `V2Transport` (encrypted on the
```
Comment (bitschmidty) [inline _posts/en/newsletters/2023-09-13-newsletter.md:68]:
```suggestion
`V1Transport` (what we have today) and `V2Transport` (encrypted on the
```

File: _posts/en/newsletters/2023-09-13-newsletter.md [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+wire). This PR, which was recently merged, is part of the
+[BIP324][bip-transport-v2]
+_Version 2 P2P Encrypted Transport Protocol_ project.
```
Comment (bitschmidty) [inline _posts/en/newsletters/2023-09-13-newsletter.md:71]: Suggest a topic link to https://bitcoinops.org/en/topics/v2-p2p-transport/

File: _posts/en/newsletters/2023-09-13-newsletter.md [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+[transport abstraction][review club 28165] is a PR by Pieter Wuille (sipa)
+that introduces a _transport_ absraction (interface class). Concrete
+derivations of this class convert a (per-peer) connection's (already
+serialized) send and receive messages to and from wire format. This
+can be thought of as implementing a deeper level of serialization and
+deserialization. These classes do not do the actual sending and receiving.
+
+The PR derives two concrete classes from the `Transport` class,
+`V1transport` (what we have today) and `V2Transport` (encrypted on the
+wire). This PR, which was recently merged, is part of the
```
Comment (bitschmidty) [inline _posts/en/newsletters/2023-09-13-newsletter.md:69]: It might be nice to work this bip324 context into the introductory sentence to help readers calibrate

File: _posts/en/newsletters/2023-09-13-newsletter.md (hunk 6 above)
Comment (bitschmidty) [reply _posts/en/newsletters/2023-09-13-newsletter.md:71]: Might also link part of the text ("project"?) to the tracking issue: https://github.com/bitcoin/bitcoin/issues/27634

File: _posts/en/newsletters/2023-09-13-newsletter.md [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+      What is the difference in what they do?"
+  a4="`CNetMsgMaker` performs the serialization of data structures
+      into bytes; `Transport` receives these bytes, adds
+      (serializes) the header, and actually sends it."
+  a4link="https://bitcoincore.reviews/28165#l-60"
+
+  q5="In the process of turning an application object like a
+      `CTransactionRef` (transaction) into bytes / network packets, what
+      happens? Which data structures does it turn into in the process?"
+  a4="`msgMaker.Make()` serializes the `CTransactionRef` message by
```
Comment (bitschmidty) [inline _posts/en/newsletters/2023-09-13-newsletter.md:113]:
```suggestion
  a5="`msgMaker.Make()` serializes the `CTransactionRef` message by
```

File: _posts/en/newsletters/2023-09-13-newsletter.md [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+      (taking that message as a simple example)?"
+  a6="36 bytes: 24 for the header (magic 4 bytes, command 12 bytes,
+      message size 4 bytes, checksum 4 bytes), then 12 bytes for the
+      payload (version 4 bytes, salt 8 bytes)."
+  a6link="https://bitcoincore.reviews/28165#l-86"
+
+  q7="After `PushMessage()` returns, have we sent the bytes corresponding
+      to this message to the peer already (yes/no/maybe)? Why?"
+  a7="All are possible. **Yes**: we (*net_processing*) don't have to do
+      anything else to get it to go. **No**: it's extremely unlikely to have
```
Comment (bitschmidty) [inline _posts/en/newsletters/2023-09-13-newsletter.md:130]: "to get it to go" could be perhaps clarified? (I dont get what it means personally)

File: _posts/en/newsletters/2023-09-13-newsletter.md [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+  a2link="https://bitcoincore.reviews/28165#l-37"
+
+  q3="What are examples of implementation bugs that could result in
+      this PR being an (accidental) consensus change?"
+  a3="A bug that restricts the maximum message size less to than
+      4MB, resulting in a blockweight soft fork; a bug in block
+      deserialization that causes the node to reject a consensus-valid
+      block, also a soft fork."
+  a3link="https://bitcoincore.reviews/28165#l-45"
+
```
Comment (harding) [inline _posts/en/newsletters/2023-09-13-newsletter.md:102]: Although instagibbs calls the <4MB theoretical bug a soft fork, neither glozow nor anyone else I see calls the generalized deserialization problem a soft fork.  I'm not sure everything in that class of failures would be a soft fork; e.g., a non-deterministic deserialization failure (kinda similar to the BIP50 failure) wouldn't be a soft fork; more relevant, the type of platform-inconsistent serialization/deserialization behavior that BIP66 surreptitiously fixed isn't something I'd have described as a soft fork if it had been triggered and caused a consensus failure.

In other words, I think we should drop the phrase ", also a soft fork" from the end of this paragraph.  (But I don't have strong feelings about it.)

File: _posts/en/newsletters/2023-09-13-newsletter.md [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+      `CTransactionRef` (transaction) into bytes / network packets, what
+      happens? Which data structures does it turn into in the process?"
+  a5="`msgMaker.Make()` serializes the `CTransactionRef` message by
+      calling `SerializeTransaction()`, then `PushMessage()` puts the
+      serialized message into the `vSendMsg` queue, then `SocketSendData()`
+      adds a header/checksum (after the changes from this PR) and asks the
+      transport for the next packet to send, and finally calls `m_sock->Send()`."
+  a5link="https://bitcoincore.reviews/28165#l-83"
+
+  q6="How many bytes are sent over the wire for the `sendtxrcncl`message
```
Comment (harding) [inline _posts/en/newsletters/2023-09-13-newsletter.md:120]: Nit: can we mention what `sendtxrcncl` message does?  I'm parsing it as SendTransactionCancel, but I doubt that's correct.

**Review by harding — COMMENTED** [review]
A couple quick notes for @LarryRuane, with the overall text looking good.

File: _posts/en/newsletters/2023-09-13-newsletter.md (hunk 10 above)
Comment (LarryRuane) [reply _posts/en/newsletters/2023-09-13-newsletter.md:102]: So, to be designated a soft fork, the behavior change in question must be intentional?

But I'm fine with dropping both mentions of "soft fork" if it's at all confusing, I think the answer will still be understandable and accurate.

File: _posts/en/newsletters/2023-09-13-newsletter.md [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+  but we note that most implementations will probably continue using one
+  set of messages for handling legacy HTLC forwarding while also
+  offering upgraded messages to support PTLC forwarding, creating two
+  different paths that will need to be maintained concurrently until
+  HTLCs are phased out.  If some implementations add experimental PTLCs
+  support before the messages are standardized, then implementations
+  might even be required to support three or more different protocols
+  simultaneously, to the disadvantage of all.
+
+    Sander's summary had not received any comments as of this writing.
```
Comment (LarryRuane) [inline _posts/en/newsletters/2023-09-13-newsletter.md:51]:
```suggestion
    Sander's summary has not received any comments as of this writing.
```
This comment is going to set the nittyness record, but the verb "had" slightly implies that it did receive comments _after_ the time of this writing.

**Comment by LarryRuane** [comment]
Force pushed to resolve Dave's comments, thanks! (Even though I asked you a follow-up question, I decided to just go ahead and make the suggested changes.)

File: _posts/en/newsletters/2023-09-13-newsletter.md (hunk 11 above)
Comment (murchandamus) [reply _posts/en/newsletters/2023-09-13-newsletter.md:121]: Yeah, it seems that it is “received from a peer demonstrating readiness to announce transactions via reconciliations.”

**Comment by LarryRuane** [comment]
Force pushed for Murch's comment, good catch, thanks!

File: _posts/en/newsletters/2023-09-13-newsletter.md (hunk 10 above)
Comment (harding) [reply _posts/en/newsletters/2023-09-13-newsletter.md:109]: @LarryRuane 

> So, to be designated a soft fork, the behavior change in question must be intentional?

Maybe?  But the point I'm trying to make is that, to be designated a soft fork, the behavior needs to be consistent.  If the behavior between nodes is inconsistent, it's a _consensus failure_.  We might eventually resolve a consensus failure using a soft fork (e.g., BIP50 resolved the BDB locks inconsistency with a soft fork that temporarily reduced the maximum block size below the level where the maximum lock count was reached, and BIP66 resolved the DER decoding inconsistency with a stricter decoding rule) but my point is that a bug in the (de)serialization code wouldn't necessarily produce a soft fork because it wouldn't necessarily occur consistently between full nodes, so it could cause a consensus failure (and consensus failures are not soft forks, at least in my personal terms sheet).

**Comment by harding** [comment]
Pushed edits for all review comments (thanks @bitschmidty @murchandamus @LarryRuane!), added lede, checked for updates to the releases/RCs (no updates), and reviewed contributions (including updates) from @LarryRuane and @murchandamus (thanks!).  Also added topic entries.  Thanks everyone!

**Review by bitschmidty — COMMENTED** [review]
ACK fixups

**Comment by bitschmidty** [comment]
Squashed and merged 🚀

File: _posts/en/newsletters/2023-09-13-newsletter.md (hunk 10 above)
Comment (LarryRuane) [reply _posts/en/newsletters/2023-09-13-newsletter.md:109]: That's an excellent way to look at it, thanks, @harding!
